tests/ops/service/storage_uploader_service_test.py

```diff
@@ -83,6 +84,63 @@ def test_uploader_s3_success_secrets_auth(
     storage_config.delete(db)
 
 
+def test_write_to_in_memory_buffer_handles_bson():
+    data = {
+        "collection:users": [
+            {
+                "paymentCustomerIds": {"stripe": "cus_abc"},
+                "birthday": datetime(1997, 1, 8, 0, 0),
+                "geolocation": {
+                    "type": "Point",
+                    "coordinates": [-122.272782, 37.871666],
+                },
+                "firstName": "Test",
+                "number": "+12345678910",
+                "name": "Test User",
+                "email": "user@example.com",
+            }
+        ],
+        "mongo_collection:purchases": [
+            {
+                "user": {"_id": ObjectId("5b4a61b1326bd9777aa61c19")},
+                "geolocation": {
+                    "type": "Point",
+                    "coordinates": [-122.41210448012356, 37.773223876953125],
+                },
+                "customerName": None,
+            },
+            {
+                "user": {"_id": ObjectId("5b4a61b1326bd9777aa61c19")},
+                "geolocation": {
+                    "type": "Point",
+                    "coordinates": [-122.4157451701343, 37.773162841796875],
+                },
+                "customerName": None,
+            },
+        ],
+        "firebase_auth:user": [
+            {
+                "provider_data": [
+                    {"email": "user@example.com", "display_name": "Test User"}
+                ],
+                "display_name": "Test User",
+                "phone_number": "+12345678910",
+                "email": "user@example.com",
+            }
+        ],
+    }
+    # This will throw a `ValueError: Circular reference detected` if no BSON
```

Comment:
ah! I was wondering where that circular reference error had been coming from.  nice work on tracking this down

Reply:
Comment seems slightly outdated now that you're no longer using BSON